frontend/packages/dev-console/src/components/monitoring/overview/MonitoringTab.tsx

```diff
@@ -1,41 +1,55 @@
 import * as React from 'react';
-import { OverviewItem } from '@console/shared';
+import { OverviewItem, usePodsWatcher } from '@console/shared';
 import { Firehose } from '@console/internal/components/utils';
-import MonitoringOverview from './MonitoringOverview';
 import { PodModel } from '@console/internal/models';
+import { PodKind } from '@console/internal/module/k8s';
+import MonitoringOverview from './MonitoringOverview';
 
 type MonitoringTabProps = {
   item: OverviewItem;
 };
 
 const MonitoringTab: React.FC<MonitoringTabProps> = ({ item }) => {
-  const { pods, monitoringAlerts } = item;
+  const { monitoringAlerts } = item;
   const {
     kind,
     metadata: { uid, name, namespace },
   } = item.obj;
-  const resources = [
-    {
-      isList: true,
-      kind: 'Event',
-      namespace,
-      prop: 'resourceEvents',
-      fieldSelector: `involvedObject.uid=${uid},involvedObject.name=${name},involvedObject.kind=${kind}`,
-    },
-  ];
+  const [pods, setPods] = React.useState<PodKind[]>([]);
+  const { podData, loadError, loaded } = usePodsWatcher(item.obj, item.obj.kind, namespace);
 
-  if (pods) {
-    pods.forEach((pod) => {
-      const fieldSelector = `involvedObject.uid=${pod.metadata.uid},involvedObject.name=${pod.metadata.name},involvedObject.kind=${PodModel.kind}`;
-      resources.push({
+  React.useEffect(() => {
+    if (!loadError && loaded) {
+      const updatedPods = podData.pods as PodKind[];
+      setPods(updatedPods);
+    }
+  }, [podData, loadError, loaded]);
```

Comment:
This doesn't need to be done inside an effect. Instead derive in place or use `useMemo`.